### PR TITLE
Add tox scripts for common commands

### PR DESCRIPTION
**Pull request by seladb**
Please let me know what you think. This PR basically add the same commands as in https://github.com/starlite-api/starlite/pull/578 but with `tox` instead of  `poetry scripts`:

- `tox` - runs tests
  - Can accept specific directories/file and will run only them, for example: `tox -- tests/caching`
  - By default, it runs on all python versions mentioned in `tox.ini`. To run it on a specific version: `tox -e py39 -- tests/caching`
- `tox -e docs` - build docs
- `tox -e lint` - runs pylint from pre-commit
  - Can accept specific directories/files and will run only them, for example: `tox -e lint starlite/cache/**`
- `tox -e fmt` - runs black, isort, prettier, blacken-docs, trailing-whitespace, end-of-file-fixer, docformatter from pre-commit
  - Can accept specific directories/files and will run only them, for example: `tox -e fmt starlite/cache/**`
- `tox -e typecheck` - runs mypy from pre-commit
  - Can accept specific directories/files and will run only them, for example: `tox -e typecheck starlite/cache/**`
- `tox -e all-checks` - runs all pre-commit checks
  - Can accept specific directories/files and will run only them, for example: `tox -e all-checks starlite/cache/**`

**NOTE:** I can add this information to [`CONTRIBUTING.md`](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.md). I haven't done it yet because I want to see if you're okay with this approach

# PR Checklist

- [x] Have you followed the guidelines in `CONTRIBUTING.md`?
- [ ] Have you got 100% test coverage on new code?
- [x] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -6,8 +6,32 @@ envlist = py37,py38,py39,py310,lint
...
+commands = pre-commit run pylint --color=always {posargs:--all-files}
+
+[testenv:fmt]
+deps = pre-commit
+ignore_errors = true
+commands =
+    pre-commit run black --color=always {posargs:--all-files}
+    pre-commit run isort --color=always {posargs:--all-files}
+    pre-commit run prettier --color=always {posargs:--all-files}
+    pre-commit run blacken-docs --color=always {posargs:--all-files}
```
Comment (seladb) [inline tox.ini:26]: Are there more pre-commit hooks that modify files (rather than just display the error/warning)?

Comment (peterschutt) [reply tox.ini:26]: The doc string one does.

**Review by peterschutt — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (seladb) [reply tox.ini:28]: I assume you mean `docformatter`?

I also added `end-of-file-fixer` and `trailing-whitespace`

**Comment by seladb** [comment]
@Goldziher are you ok with adding this?
If yes I'll add usage info to [CONTRIBUTING.md](https://github.com/starlite-api/starlite/blob/main/CONTRIBUTING.md) and merge

**Review by cofin — APPROVED** [review]

**Pull request merged**